<launch>
	
	<node pkg="ohm_bob_laser_filter" type="bob_laser_filter" name="bob_laser_filter">
		<param name = "laser_output_topic" type = "string" value = "filtered_scan"/>
	<param name="min_angle" type="double" value="-2.0"/>
	<param name="max_angle" type="double" value="2.0"/>
	<param name="max_range" type ="double" value ="60.0"/>	
		<param name="mirror" type = "bool" value = "true"/>
  </node>
  
  <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server" output="screen">
    <param name="source_frame_name" type="string" value="laser"/>
  </node>
  <node name = "slam_node" pkg = "ohm_tsd_slam" type = "localize_node" output = "screen">
         <!-- topic parameters -->
         <param name = "map_topic"               type = "string" value = "/map"  />    <!-- topic of occupancy grid              -->
         <param name = "pose_topic"              type = "string" value = "pose"  />    <!-- topic for robot's pose               -->   
         <param name = "laser_topic"             type = "string" value = "filtered_scan"  />    <!-- topic to receive laser scan          -->
         
         <param name ="data_path" type = "string" value = "/home/phil/.ros/tsd_grid.dat"/>
         <!-- map parameters -->
         <param name = "x_offset"            type = "double" value = "0.0"   />    <!-- offset of robot in map in meters     -->
         <param name = "y_offset"            type = "double" value = "0.0"   />    <!-- offset of robot in map in meters     -->
         <param name = "yaw_offset"              type = "double" value = "0.0"   />    <!-- offset of yaw                        -->
         
         <param name = "map_size"      type = "int"    value = "13"    />    <!-- map size                             -->
         <param name = "cellsize"                type = "double" value = "0.02"  />    <!-- cell size in meters                  -->
         <param name = "truncation_radius"       type = "int"    value = "5"     />    <!-- radius as factor of cell size        -->
         <param name = "occ_grid_time_interval"  type = "double" value = "2.0"   />    <!-- frequency to publish map             -->
         <param name = "object_inflation_factor" type = "int"    value = "0"     />    <!-- factor to inflate obstacles          -->

         <!-- raycasting parameters -->
         <param name = "min_range"               type = "double" value = "0.0"   />    <!-- minimum range of raycaster in meters -->
         <param name = "max_range"               type = "double" value = "60.0"  />    <!-- maximum range of raycaster in meters -->
         <param name = "low_reflectivity_range"  type = "double" value = "3.0"   />
         
            
         <param name = "loop_rate"               type = "double" value = "100.0" />
         <param name = "tf_base_frame"           type = "string" value = "/map"  />
         <param name = "tf_child_frame"          type = "string" value = "laser" />
         <param name = "use_object_inflation"    type = "bool"   value = "false" />
         <param name = "get_map_topic"           type = "string" value = "/map"  />
         
         <!-- robot parameters -->
         <param name = "footprint_width"         type = "double" value = "1.0"   />    <!-- width of robot in meters             -->
         <param name = "footprint_height"        type = "double" value = "1.0"   />    <!-- length of robot in meters            -->
         <param name = "footprint_x_offset"      type = "double" value = "-0.5"  />    <!-- offset of laser in meters            -->
  
         <!-- registration parameters -->	
         <param name = "registration_mode"       type = "int"    value = "0"     />    <!-- 0: ICP    1: EXPERIMENTAL -->	
         	
         <!-- ICP parameters -->
         <param name = "dist_filter_max"         type = "double" value = "1.0"   />    <!-- maximum distance for matching points -->
         <param name = "dist_filter_min"         type = "double" value = "0.01" />    <!-- minimum distance for matching points -->
         <param name = "icp_iterations"          type = "int"    value = "100"    />    <!-- number of icp iterations             -->
         <param name = "reg_trs_max"             type = "double" value = "1.0"   />    <!-- maximum translation between two scans-->
         <param name = "reg_sin_rot_max"         type = "double" value = "0.5"   />    <!-- maximum rotation between two scans   -->
         
         <!-- ransac parameters -->
         <param name = "ransac_trials"           type = "int"    value = "50"    />    <!-- number of ransac trials              -->
         <param name = "ransac_eps_thresh"       type = "double" value = "0.15"  />    <!-- threshold for outlier rejection in meters -->
         <param name = "ransac_ctrlset_size"     type = "int"    value = "180"   />    <!-- number of points taking for ransac   -->
         <param name = "ransac_phi_max"          type = "double" value = "45.0"  />    <!-- maximum angle for ransac matching    -->
         <param name = "call_hive"               type = "bool"   value = "false"/>         
      </node>
	
  <!--<node name = "localize_node" pkg = "ohm_tsd_slam" type = "localize_node" output = "screen">
  	  <param name ="data_source"              type = "string" value="/home/phil/.ros/tsd_grid.dat"/>
  	  <param name ="data_path" trype = "string" value = "/home/user/.ros/tsd_grid.dat"/>
      <param name = "laser_topic"             type = "string" value = "filtered_scan"  />
      <param name = "x_offset"            type = "double" value = "0.0"   />
      <param name = "y_offset"            type = "double" value = "0.0"   />
	  <param name = "yaw_offset"              type = "double" value = "0.0"   />      
      <param name = "min_range"               type = "double" value = "0.0"   />
      <param name = "max_range"               type = "double" value = "30.0"  />
      <param name = "low_reflectivity_range"  type = "double" value = "3.0"   />
      <param name = "occ_grid_time_interval"  type = "double" value = "2.0"   />
      <param name = "loop_rate"               type = "double" value = "40.0"  />
      <param name = "pose_topic"              type = "string" value = "pose"  />
      <param name = "tf_base_frame"           type = "string" value = "/map"  />
      <param name = "tf_child_frame"          type = "string" value = "laser" />
      <param name = "truncation_radius"       type = "double" value = "2.0"   />
      <param name = "object_inflation_factor" type = "int"    value = "0"     />
      <param name = "use_object_inflation"    type = "bool"   value = "false" />
      <param name = "robot_length"            type = "double" value = "0.75"  />
      <param name = "robot_width"             type = "double" value = "0.5"   /> 
      <param name = "map_topic"               type = "string" value = "/map"  />
      <param name = "get_map_topic"           type = "string" value = "/map"  /> 
      <param name = "footprint_width"         type = "double" value = "0.15"  />	
	  <param name = "footprint_height"        type = "double" value = "0.15"  />
	  <param name = "call_hive"               type = "bool"   value = "false"/>  
	</node>-->
</launch>
